SLAM/iterative_closest_point/iterative_closest_point.py

```diff
@@ -48,15 +49,20 @@ def icp_matching(previous_points, current_points):
         indexes, error = nearest_neighbor_association(previous_points, current_points)
         Rt, Tt = svd_motion_estimation(previous_points[:, indexes], current_points)
 
+        dError = preError - error
+        print("Residual:", error)
+
+        if dError < 0:  # prevent matrix H changing, exit loop
+            print("Not Converge...", preError, dError, count)
+            break
+
+        preError = error
+
         # update current points
         current_points = (Rt @ current_points) + Tt[:, np.newaxis]
 
         H = update_homogeneous_matrix(H, Rt, Tt)
 
-        dError = abs(preError - error)
-        preError = error
-        print("Residual:", error)
-
         if dError <= EPS:
```

Comment:
We can merge these break statements to line 55 like:

```
if 0 <= dError <= EPS:
    print("Converge", error, dError, count)
    break
elif dError < 0:
    print("Not Converge... Error increased. ", preError, dError, count)
    break
elif MAX_ITER <= count:
     print("Not Converge... Reached max iterations", error, dError, count)
     break
```

Reply:
I cannot see how we can merge break statements. Because it's important for ```H = update_homogeneous_matrix``` to be after ```dError < 0``` (as at this break we take H status from previous iteration) and before ```dError <= EPS``` (as we need to update H before possible breaking for these 2 cases).

P.S. I thought it over, it is possible to move up 2 last break statements, but there will always be something between ```dError < 0``` and the others. Moving them up, it is necessary to do ```dError = preError - error``` after them but before ```dError < 0```. To my mind it becomes a mess in algorithm logic, so I suggest to keep code as is.
I share your wish to make things beautiful. But here I don't see a way to make it better.

Reply:
OK. Make sense. Thanks.